source/grid/tria.cc

```diff
@@ -11985,6 +11985,21 @@ Triangulation<dim, spacedim>::last_active() const
 
 
 
+template <int dim, int spacedim>
+typename Triangulation<dim, spacedim>::cell_iterator
+Triangulation<dim, spacedim>::create_cell_iterator(const CellId &cell_id) const
+{
+  cell_iterator cell(
+    this, 0, coarse_cell_id_to_coarse_cell_index(cell_id.get_coarse_cell_id()));
+
+  for (const auto &child_index : cell_id.get_child_indices())
+    cell = cell->child(static_cast<unsigned int>(child_index));
```

Comment:
I think this would be the place where it's worth adding an assertion that `cell->has_children() == true` and if not provide an error message along the lines of what I suggested to put into the documentation.